Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/data/LocalDataSource.kt

```diff
@@ -0,0 +1,53 @@
+package com.mentalmachines.droidcon_boston.data
+
+import android.content.Context
+import com.mentalmachines.droidcon_boston.modal.Media
+import com.mentalmachines.droidcon_boston.modal.QuotedTweet
+import com.mentalmachines.droidcon_boston.modal.Tweet
+import com.mentalmachines.droidcon_boston.modal.TweetWithMedia
+
+class LocalDataSource private constructor(private val appDatabase: AppDatabase) : DataSource {
+
+    companion object {
+        private lateinit var dataSource: LocalDataSource
+        fun getInstance(context: Context): LocalDataSource {
+            if (!::dataSource.isInitialized) {
```

Comment:
Is it possible to use the lazy delegate here instead? `private var dataSource: LocalDataSource by lazy { ... }`

Reply:
getInstance is a static factory method which generates an instance of LocalDataSource. How about this ?

```suggestion
            private var dataSource: LocalDataSource? = null
            
            return dataSource ?: LocalDataSource(AppDatabase.getInstance(context)).also { dataSource = it }
```

Reply:
Just like I did for AppDatabase.